src/parser.rs

```diff
@@ -6,22 +6,14 @@ use nom::{
     bytes::streaming::take,
     combinator::map,
     number::streaming::{be_i16, be_i24, be_i32, be_i8, be_u16, be_u24, be_u32, be_u8},
-    IResult, Needed,
+    IResult,
 };
 
-pub fn parse(i: &[u8]) -> IResult<&[u8], Instruction> {
-    match i.get(0) {
-        Some(&op) if op <= 127 => Ok((&i[1..], Instruction::Set(op as u32))),
-        Some(&op) if op >= 171 && op <= 234 => Ok((&i[1..], Instruction::Font((op - 171) as u32))),
-        Some(&_) => parse_complex(i),
-        None => Err(nom::Err::Incomplete(Needed::Unknown)),
```

Comment:
With these code changes, the error returned if the input is empty will instead be `nom::Err::Incomplete(Needed::Size(1))` via the call to `be_u8`. I think strictly speaking this is a more accurate error. However it is an API change so happy not to do that.